File: src/app/shared/service/api/authority.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpRequest, HttpParams } from '@angular/common/http';
import { AuthorityDTORequest } from '../../model/request/authorityDTORequest';
import { environment } from 'src/environments/environment';
import { Observable } from 'rxjs';

@Injectable({
    providedIn: 'root'
})
export class AuthorityService {
    constructor(
        private _httpClient: HttpClient
    ) { }

    save(dto: AuthorityDTORequest) {
        const requ = new HttpRequest('POST', `${environment.urlBase}/authorities/authority`, dto, {
            reportProgress: true,
            responseType: 'json'
        });
        return this._httpClient.request(requ);
    }

    public update(dto: AuthorityDTORequest, id: number) {
        const requ = new HttpRequest('PUT', `${environment.urlBase}/authorities/authority/` + id, dto, {
            reportProgress: true,
            responseType: 'json'
        });
        return this._httpClient.request(requ);
    }

    public delete(id: number) {
        const requ = new HttpRequest('PATCH', `${environment.urlBase}/authorities/authority/` + id, {
            reportProgress: true,
            responseType: 'json'
        });
        return this._httpClient.request(requ);
    }

    findById(id: number): Observable<any> {
        return this._httpClient.get<any>(environment.urlBase + '/authorities/authority/' + id);
    }

    findByIduser(id: number, page: number, size: number): Observable<any> {
        let queryParameters = new HttpParams();
        queryParameters = queryParameters.append('page', page);
        queryParameters = queryParameters.append('size', size);
        queryParameters = queryParameters.append('field', '');
        queryParameters = queryParameters.append('order', '');
        return this._httpClient.get<any>(environment.urlBase + '/authorities/authority/user/' + id + '?' + queryParameters);
    }

    findByName(name: string, page: number, size: number): Observable<any> {
        let queryParameters = new HttpParams();
        queryParameters = queryParameters.append('page', page);
        queryParameters = queryParameters.append('size', size);
        queryParameters = queryParameters.append('field', '');
        queryParameters = queryParameters.append('order', '');
        queryParameters = queryParameters.append('name', name ? name : '');
        return this._httpClient.get<any>(environment.urlBase + '/authorities/authority?' + queryParameters);
    }
}

File: src/main/java/com/sys/botica/crce/pe/sys_botica/service/AuthorityService.java
package com.sys.botica.crce.pe.sys_botica.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.sys.botica.crce.pe.sys_botica.dto.AuthorityDTO;
import com.sys.botica.crce.pe.sys_botica.dto.HrefEntityDTO;
import com.sys.botica.crce.pe.sys_botica.dto.request.AuthorityDTORequest;

public interface AuthorityService {
	public Page<AuthorityDTO> findByName(String name,Pageable pageable);
	
	public HrefEntityDTO update(AuthorityDTORequest dto,Long id);
	
	public HrefEntityDTO delete(Long id);
	
	public HrefEntityDTO save(AuthorityDTORequest dto);
	
	public AuthorityDTO findById(Long id);
	
	public Page<AuthorityDTO> findAllByUser(Long iduser,Pageable pageable);
}
